test/zero-downtime-flagd-proxy/main.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	pb "buf.build/gen/go/open-feature/flagd/grpc/go/sync/v1/syncv1grpc"
+	"google.golang.org/grpc"
+)
+
+func main() {
+	waitSecondsBetweenRequests := getWaitSecondsBetweenRequests()
+	flagdURL := getURL()
+
+	// Create a channel to receive a signal when the gRPC connection fails
+	errChan := make(chan bool)
+
+	// Use a goroutine to run your program logic
+	go func() {
+		if err := handleRequests(waitSecondsBetweenRequests, flagdURL); err != nil {
+			errChan <- true
+		}
+	}()
+
+	// The program should run until it receives an error
+	<-errChan
+}
+
+func handleRequests(waitSecondsBetweenRequests int, flagdURL string) error {
+	var conn *grpc.ClientConn
+	var grpcClient pb.FlagSyncServiceClient
+	// open the connection only once
+	conn, grpcClient = establishGrpcConnection(flagdURL)
+
+	defer func() {
+		if conn != nil {
+			// clean up
+			err := conn.Close()
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+		}
+	}()
+
+	return doRequests(grpcClient, waitSecondsBetweenRequests)
+}
+
+func getWaitSecondsBetweenRequests() int {
+	return getEnvVarOrDefault("WAIT_TIME_BETWEEN_REQUESTS_MS", 1)
```

Comment:
here we have a slight mismatch, as the name of the env var suggests that the wait time is provided in milliseconds, however that value is treaded as seconds in the code